lava-docs: add controls section (#4)
adevinta/lava-docs: src/controls.md

> # Controls and customization

Reviewer: ```suggestion
# Security Controls
```

> Behind the scenes Lava makes use of other security tools and custom security

Reviewer: ```suggestion
Behind the scenes Lava uses other security tools and custom security
```

checks to perform a security scan. Tools are integrated through what is called
["checktypes"](helpdoc/checktypes.md).

Checktypes are part of the Vulcan ecosystem and both Vulcan and Lava share the

> same collection located at <https://github.com/adevinta/vulcan-checks>

Reviewer: ```suggestion
same collection located at <https://github.com/adevinta/vulcan-checks>.
```

> ## Type of controls

Reviewer: ```suggestion
## Types of security controls
```

> This table classifies each checktype based on the security controls that it
> covers.

Reviewer: ```suggestion
The following table lists all the security controls covered by Lava's
default checktype collection.
```

> | <div style="width:175px">Control</did> | Description                                                                                                                                                         |

Reviewer: I would use mdbook's default style and rename the column to `Security Control`.

```suggestion
| Security Control | Description                                                                                                                                                         |
```

| SAST                                   | Static Application Security Testing.                                                                                                                                |
| SAST - Application                     | Static analysis of application-related assets. Inother words, security analysis of the source code of an application.                                               |
| SAST - IaC                             | Static analysis of Infrastructure as Code manifests. For instance, AWS CloudFormation templates or Terraform configurations.                                        |
| SCA                                    | Software Composition Analysis. For instance, detection of outdated dependencies.                                                                                    |
| Secret detection                       |                                                                                                                                                                     |

> | Finding collection                     | This category referees to checktypes that do not run the analysis themselves. Instead, they collect the findings from 3rd party systems like Tenable.io or vulners. |

Reviewer: ```suggestion
| Finding collection                     | This category refers to checktypes that do not run the analysis themselves. Instead, they collect the findings from 3rd party systems like Tenable.io or Vulners. |
```

| Recon                                  | Checktypes focused on reconnaissance activities. For instance, port scanning and service discovery.                                                                 |

> | Data analysis                          | Checktypes focused on processing already available data to produce results from the point of view of information security.                                          |

Reviewer: ```suggestion
| Data analysis                          | Checktypes focused on processing data sets to extract information considered relevant from an information security point of view.                                          |
```

> 

Reviewer: Remove extra newline.

```suggestion
```

> ## Enabled by Default

Reviewer: ```suggestion
## Enabled by default
```

<table>
  <thead>
    <tr>
      <th>Checktype</th>
      <th>Description</th>

>       <th>Type of Control</th>

Reviewer: ```suggestion
      <th>Security Control</th>
```

    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <a href="https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-nuclei">vulcan-nuclei</a>
      </td>
      <td>
        Scan web addresses with projectdiscovery/nuclei.<br>
        Official Documentation:
        <a href="https://docs.projectdiscovery.io/tools/nuclei"
          >https://docs.projectdiscovery.io/tools/nuclei</a

>         >

Reviewer: I would use markdown tables and single-line descriptions. E.g.

```
Scan web addresses with projectdiscovery/nuclei. Official Documentation: <https://docs.projectdiscovery.io/tools/nuclei>.
```

        Official Documentation:
        <a href="https://semgrep.dev/docs">https://semgrep.dev/docs</a>
      </td>
      <td>SAST - Application</td>
    </tr>
    <tr>
      <td>
        <a href="https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-trivy">vulcan-trivy</a>
      </td>
      <td>
        Scan docker images and Git repositories using aquasec/trivy.<br>
        Official Documentation:
        <a href="https://aquasecurity.github.io/trivy/latest"
          >https://aquasecurity.github.io/trivy/latest</a
        >
      </td>
      <td>SCA, Secret Detection, SAST - IaC</td>
    </tr>
    <tr>
      <td>
        <a href="https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-zap">vulcan-zap</a>
      </td>
      <td>
        Runs an OWASP ZAP passive or active scan.<br>
        Official Documentation:
        <a href="https://www.zaproxy.org/docs">https://www.zaproxy.org/docs</a>
      </td>
      <td>DAST - Application</td>
    </tr>
  </tbody>
</table>

Since the check versions are updated regularly in an automatic way, the best way
to know the version that Lava is using is checking the source code of each
check.

> ## Disabled by Default

Reviewer: ```suggestion
## Disabled by default
```

> | Checktype                                                                                                            | Description                                                                             | Type of Control                     |

Reviewer: ```suggestion
| Checktype                                                                                                            | Description                                                                             | Security Control                     |
```

| [vulcan-semgrep](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-semgrep)   | Finds potential issues in source code from a Git repository using Semgrep. Official Documentation: <https://semgrep.dev/docs>      | SAST - Application                |
| [vulcan-trivy](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-trivy)       | Scan docker images and Git repositories using aquasec/trivy. Official Documentation: <https://aquasecurity.github.io/trivy/latest> | SCA, Secret Detection, SAST - IaC |
| [vulcan-retirejs](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-retirejs) | Check web pages for vulnerable JavaScript libraries. Official Documentation: <https://retirejs.github.io/retire.js>                | SCA                               |
| [vulcan-nuclei](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-nuclei)     | Scan web addresses with projectdiscovery/nuclei. Official Documentation: <https://docs.projectdiscovery.io/tools/nuclei>           | DAST - Application                |

> | [vulcan-zap](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-zap)           | Runs an OWASP ZAP passive or active scan. Official Documentation: <https://www.zaproxy.org/docs>                                   | DAST - Application                |                                                                                 |

Reviewer: ```suggestion
| [vulcan-zap](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-zap)           | Runs an OWASP ZAP passive or active scan. Official Documentation: <https://www.zaproxy.org/docs>                                   | DAST - Application                |
```

## Type of security controls

The following table lists all the security controls covered by Lava's
checktype collection.

| Security Control   | Description                                                                                                                                                       |
|--------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| SAST               | Static Application Security Testing. It's the analysis of source code.                                                                                            |
| SCA                | Software Composition Analysis. For instance, detection of outdated dependencies.                                                                                  |

> | IaC Security       | Static analysis of Infrastructure as Code manifests. For instance, AWS CloudFormation templates or Terraform configurations.                                      |

Reviewer: For consistency with the other security controls.

```suggestion
| IaC security       | Static analysis of Infrastructure as Code manifests. For instance, AWS CloudFormation templates or Terraform configurations.                                      |
```

| Data analysis      | Checktypes focused on processing data sets to extract information considered relevant from an information security point of view.                                 |

## Enabled by default

| Checktype                                                                                    | Description                                                                                                                        | Security Control                    |
|----------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------|-------------------------------------|
| [vulcan-semgrep](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-semgrep)   | Finds potential issues in source code from a Git repository using Semgrep. Official Documentation: <https://semgrep.dev/docs>      | SAST                                |

> | [vulcan-trivy](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-trivy)       | Scan docker images and Git repositories using aquasec/trivy. Official Documentation: <https://aquasecurity.github.io/trivy/latest> | SCA, Secret Detection, IaC Security |

Reviewer: For consistency with the previous table.

```suggestion
| [vulcan-trivy](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-trivy)       | Scan docker images and Git repositories using aquasec/trivy. Official Documentation: <https://aquasecurity.github.io/trivy/latest> | SCA, IaC security, Secret detection |
```

| [vulcan-retirejs](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-retirejs) | Check web pages for vulnerable JavaScript libraries. Official Documentation: <https://retirejs.github.io/retire.js>                | SCA                                 |
| [vulcan-nuclei](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-nuclei)     | Scan web addresses with projectdiscovery/nuclei. Official Documentation: <https://docs.projectdiscovery.io/tools/nuclei>           | DAST                                |
| [vulcan-zap](https://github.com/adevinta/vulcan-checks/tree/master/cmd/vulcan-zap)           | Runs an OWASP ZAP passive or active scan. Official Documentation: <https://www.zaproxy.org/docs>                                   | DAST                                |

Since the check versions are updated regularly in an automatic way, the best way
to know the version that Lava is using is checking the source code of each
check.

## Disabled by default

> | Checktype                                                                                                            | Description                                                                             | Security Control   |

Reviewer: Use the same case as in the first table (i.e. `Data analysis`, `Finding collection`, `Secret detection`).
